core/src/com/unciv/ui/worldscreen/WorldScreen.kt

```diff
@@ -242,10 +242,8 @@ class WorldScreen(val gameInfo: GameInfo, val viewingCiv:CivilizationInfo) : Cam
             val latestGame = OnlineMultiplayer().tryDownloadGame(gameInfo.gameId)
 
             // if we find it still isn't player's turn...nothing changed
-            if (gameInfo.currentPlayer == latestGame.currentPlayer) {
+            if (viewingCiv.civName != latestGame.currentPlayer) {
                 Gdx.app.postRunnable { loadingGamePopup.close() }
-                // edge case - if there's only one player in a multiplayer game, we still check online, but it could be that we were correct and it is our turn
```

Comment:
Are you sure this still works for 'single human player' in a multiplayer game?

Reply:
Yep works fine.
Tested it with only one human player.
I just found another bug. If the upload fails the own turn gets reset because "shouldUpdate" gets set to true instead of false. I don't think we want the game to update if the upload fails. Changed with the next commit.

Reply:
The game will update anyway the next time the layer clicks anything - if there's a problem with the update then we should deal with it in the update function
In any case there's usually no good reason to set shouldUpdate to false, if it's not needed we should just remove the line

Reply:
I just reverted that for now because I realized it doesn't really fix it, just for specific circumstances. And I don't want to touch anything that I don't need to touch. I will probably come along and change it once I found the real problem that causes the turn to reset. But it's hard to replicate.